internal/graph/cached_resolver_test.go

```diff
@@ -654,7 +654,7 @@ func TestCachedCheckDatastoreQueryCount(t *testing.T) {
 	ttuLocalChecker.Close()
 
 	require.NoError(t, err)
-	require.Equal(t, uint32(1), res.GetResolutionMetadata().DatastoreQueryCount)
+	require.Equal(t, uint32(2), res.GetResolutionMetadata().DatastoreQueryCount)
```

Comment:
?

Reply:
still need to figure this one out

Reply:
It's because: in the old slow way, we divide the TTU check into subproblems. 

Eg if the query is `Check(user:maria, viewer, document:1)`

1. We find `folder:x` is parent of `document:1`
2. Do`Check(user:maria, viewer, folder:x)`. This is stored in the internal cache thanks to the call to `dispatch` here: https://github.com/openfga/openfga/blob/d32341e622f63cbaf5ff0fb384f3e2af93cd7975/internal/graph/check.go#L687

In the new way we do:
1. We find `folder:x` is parent of `document:1`
2. Find all `folder`s that` user:maria` can view. This part we are not caching, and I don't see how we'd do that, because it's not a Check subproblem (ie the result is not a bool, is a list)

Reply:
I had a discussion with Yissell on actually caching the results of the short path in order to also optimize recursion of usersets. We can discuss that as an N+1.

Reply:
@elbuo8 would that be changing the caching pattern to, instead of caching Check queries, cache the underlying datastore requests?

Reply:
OK. We can extend https://github.com/openfga/openfga/blob/3540b66777b2983e0f37581775759a4a5989b6d5/pkg/storage/storagewrappers/caching.go#L19-L23 for that